eng/scripts/CodeCheck.ps1

```diff
@@ -190,6 +190,31 @@ try {
             & git --no-pager diff --ignore-space-change $filePath
         }
     }
+
+    Write-Host "Checking for changes to API baseline files"
+
+    # Retrieve the set of changed files compared to main
+    $commitSha = git rev-parse HEAD
+    $changedFilesFromMain = git --no-pager diff origin/main...$commitSha --ignore-space-change --name-only
```

Comment:
Don't need to make the change now, but if there's a way to not hard-code the branch name that'd be preferable

Reply:
I'm not sure how we can remove the hard coding to the default branch, is there a git command to do that? In any case, I figured the default branch name shouldn't be modified often and I think in those cases, we would scrub the script for usages of the name anyway.

Reply:
There be dragons @JunTaoLuo In particular,
1. We need to enable a more drastic version of this in `release/5.0`, covering the  PublicAPI.Unshipped.txt files too
    - We need to compare against `release/5.0` there
2. Need to compare against our preview branches for 6.0 after each is split from `main`
3. This check will do nothing in rolling builds and should be suppressed unless `$(Build.Reason)` (in YAML) or `$env:BUILD_REASON` contain `PullRequest`

To get _current_ branch in PR builds, can use

- `git config --show-current` displays the current branch name
- same information should be available in `$(Build.SourceBranchName)` in YAML or `$env:BUILD_SOURCEBRANCHNAME`
- use above w/o `Name` / `NAME` to get the full branch path but, with the name, should display the PR id

Need something like the GitHub REST API to get more information e.g. to display the base aka target branch
``` powershell
Invoke-WebRequest -Headers @{ Accept = "application/vnd.github.v3+json"} https://api.github.com/repos/dotnet/aspnetcore/pulls/{PR id} | `
 ConvertFrom-Json |ForEach-Object `
 base |ForEach-Object `
 ref
```

For internal PRs, the commands are probably like
``` powershell
Invoke-WebRequest  -Headers @{ Accept = "application/vnd.github.v3+json"} https://dev.azure.com/{blah}/{blah}/_git/dotnet-aspnetcore/_apis/git/pullrequests/{PR id}?api-version=5.0 |ForEach-Object `
  targetRefName
```

But, I don't see documentation on how to authenticate nor control the preferred response type. (May respond w/ JSON in successful cases but I'm only getting 404s despite testing w/ an existing PR.)

@ilyas1974 anyone in your world with the AzDO REST API knowledge we need❔ Note we'd like to run this in CI, where we should have the right tokens available.

Reply:
Suepct we'd be fine w/ a translation of https://docs.microsoft.com/en-us/rest/api/azure/devops/git/pull%20requests/get%20pull%20request%20by%20id?view=azure-devops-rest-5.0#security into English 😀